### ISSUE TITLE
Adaptive theme support for Linux

### ISSUE THREAD

**Issue by swsnr** [raphamorim/rio#408]
Linux provides a standard cross-desktop API to query the system color scheme, see <https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.Settings.html>.  ashpd provides a Rust library for these portals, see https://bilelmoussaoui.github.io/ashpd/ashpd/desktop/settings/index.html for access to settings.

I wanted to make a pull request but haven't been able to figure out where the theme information comes from on Windows/macOS, and I wanted to clarify whether a new dependency would be okay.

**Comment by raphamorim** [comment]
Hey @swsnr thanks for the issue, sorry for the delay.

Assuming would be implemented by our side I think is ok to add a new dependency but I think now this should be winit responsability since they provide theming support, do you know if winit does cover it?

**Comment by swsnr** [comment]
I don't think it does.  It's doc.rs page lists no dependencies that'd support this, so I doubt it has access to the portal API. 

And I don't think it would, either.  winit would have to add pretty heavy-weight dependencies on either zbus or gio to access the portal API over DBus; it think that's somewhat out of scope for a small windowing toolkit.

**Comment by swsnr** [comment]
I am not using rio so I'm taking the liberty to close this ticket as I am no longer interested in this feature.

**Comment by AdsonCicilioti** [comment]
I was very excited to start testing Rio, but without this feature I'm going to abort.

I just know that the main Desktop Envinroments and even TWMś already offer and issue this color change event and even websites with this functionality follow this change, like the Github website itself, for example. It looks like they follow a variable called prefers-color-scheme`

**Comment by raphamorim** [comment]
Yea I think this issue is important and bring value to linux users. I will keep it open and try to prioritize to work in the future, but if anyone would like to work on it pls ping here or on our discord

**Comment by AdsonCicilioti** [comment]
@raphamorim Maybe if there was an option flag to change the theme on the fly it would be very welcome, so we could easily automate the change between light and dark themes via script.

**Comment by pinpox** [comment]
I have implemented this feature in https://github.com/raphamorim/rio/pull/1358

It works on my machine (wayland), feel free to review and request changes as needed.

**Comment by raphamorim** [comment]
Thanks all!
